optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,862 @@
+                if (meetingAssignmentList.isEmpty() && mergingPreviousTimeGrain && previousMeetingRemainingTimeGrains > 0) {
+                    previousMeetingRemainingTimeGrains--;
+                    nextCellVertically();
+                } else {
+                    if (mergingPreviousTimeGrain && mergeStart < currentRowNumber) {
+                        currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, currentColumnNumber, currentColumnNumber));
+                    }
+
+                    StringBuilder meetingInfo = new StringBuilder();
+                    for (MeetingAssignment meetingAssignment : meetingAssignmentList) {
+                        String startTimeString = getTimeString(meetingAssignment.getStartingTimeGrain().getStartingMinuteOfDay());
+                        String endTimeString = getTimeString(solution.getTimeGrainList().get(meetingAssignment.getLastTimeGrainIndex()).getStartingMinuteOfDay()
+                                                                 + TimeGrain.GRAIN_LENGTH_IN_MINUTES);
+                        meetingInfo.append(StringUtils.abbreviate(meetingAssignment.getMeeting().getTopic(), 150)).append("\n  ")
+                            .append(meetingAssignment.getMeeting().getSpeakerList().stream().map(Person::getFullName).collect(joining(", "))).append("\n  ")
+                            .append(startTimeString).append(" - ").append(endTimeString)
+                            .append(" (").append(meetingAssignment.getMeeting().getDurationInGrains() * TimeGrain.GRAIN_LENGTH_IN_MINUTES).append(" mins)");
+                    }
+                    nextCellVertically().setCellValue(meetingInfo.toString());
+
+                    previousMeetingRemainingTimeGrains = getLongestDurationInGrains(meetingAssignmentList) - 1;
+                    mergingPreviousTimeGrain = previousMeetingRemainingTimeGrains > 0;
+                    mergeStart = currentRowNumber;
+                }
+            }
+            if (mergeStart < currentRowNumber) {
+                currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, currentColumnNumber, currentColumnNumber));
+            }
+        }
+
+        private String getTimeString(int minuteOfDay) {
+            return TIME_FORMATTER.format(LocalTime.ofSecondOfDay(minuteOfDay * 60));
+        }
+
+        private void writeMeetingAssignmentList(List<MeetingAssignment> meetingAssignmentList) {
+            String[] filteredConstraintNames = {
+                DO_ALL_MEETINGS_AS_SOON_AS_POSSIBLE, MINIMUM_TIMEGRAINS_BREAK, OVERLAPPING_MEETINGS, ASSIGN_LARGER_ROOMS_FIRST,
+
+                REQUIRED_AND_PREFERRED_ATTENDANCE_CONFLICT, PREFERRED_ATTENDANCE_CONFLICT, MINIMUM_TIMEGRAINS_BREAK,
+
+                ROOM_CONFLICT, DONT_GO_IN_OVERTIME, REQUIRED_ATTENDANCE_CONFLICT, REQUIRED_ROOM_CAPACITY, START_AND_END_ON_SAME_DAY,
+                ENTIRE_GROUP_MEETING_NOT_SCHEDULED};
```

Comment:
`ROOM_STABILITY` is missing and `MINIMUM_TIMEGRAINS_BREAK` is there twice. I would recommend to to move the constraint constants into a enum class. You could then obtain the complete set of constraints by calling `MeetingSchedulingConstraints.values()` and easily remove room stability if it's supposed to be filtered out. So whenever you add a new item to the enum class, you'll can get the complete set (or minus one).

If you want to remove the maintenance risk completely, you can add a test that will build a kbase from the DRL file, extract all rules and check that all rules are enumerated.